client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx

```diff
@@ -30,11 +30,11 @@ function getUrlInfo( url: string ) {
 	const urlWithoutProtocol = url.replace( /^https?:\/\//, '' );
 
 	// Ex. mytest.wordpress.com matches mytest
```

Comment:
Worth noting that in my view, the types should have been correct before the change in this PR, but TypeScript seems to be having a hard time inferring the types. 

`const siteName = urlWithoutProtocol.match( /^[^.]*/ );` returns a type of `RegExpMatchArray`. Technically, IF that array exists and you grab the first item of that array `(`siteName[0]`)`, it should always be a string. If not defined, it should have been an empty string. But TS was still reading it as potentially undefined. 

<img width="1006" alt="sidebar-ts-error" src="https://user-images.githubusercontent.com/21228350/221700647-1c86b639-f353-41d7-a92c-46330c460711.png">

Reply:
As a small side note, we use let to define some vars in this file, and then update them later based on sidebarDomain. The fact that we define and then update them made me think we should be using useState. But this is technically just derived state. The values only changes based on changes in props. Even so, the way we're doing it is a bit odd. Normally, for derived state, you should be able to just define the values once with const.